Monika After Story/game/definitions.rpy

```diff
@@ -662,41 +671,72 @@ python early:
             current_time = datetime.datetime.now()
             # insertion sort
             for ev in ev_list:
-                event_time = True
+
+                e = events[ev]
 
                 #If the event has no time-dependence, don't check it
-                if (events[ev].start_date is None) and (events[ev].end_date is None):
-                    event_time = False
+                if (e.start_date is None) and (e.end_date is None):
+                    continue
 
                 #Calendar must be based on a date
-                if events[ev].start_date is not None:
-                    if events[ev].start_date > current_time:
-                        event_time = False
+                if e.start_date is not None:
+                    if e.start_date > current_time:
+                        continue
 
-                if events[ev].end_date is not None:
-                    if events[ev].end_date <= current_time:
-                        event_time = False
+                if e.end_date is not None:
+                    if e.end_date <= current_time:
+                        continue
 
-                if events[ev].conditional is not None:
-                    if not eval(events[ev].conditional):
-                        event_time = False
+                if e.conditional is not None:
+                    if not eval(e.conditional):
+                        continue
 
 
-                if event_time and events[ev].action is not None:
+                if e.action is not None:
                     #Perform the event's action
-                    if events[ev].action == EV_ACT_PUSH:
+                    if e.action == EV_ACT_PUSH:
                         pushEvent(ev)
                     elif events[ev].action == EV_ACT_QUEUE:
```

Comment:
@aldoram5 there's an inconsistency here, why is this line not using `e.action`

Reply:
fixed, I need to get a new pair of glasses ~~That's true though broke them yesterday~~